kratos/future/solving_strategies/strategies/implicit_strategy.h

```diff
@@ -0,0 +1,750 @@
+     */
+    bool GetReformDofsAtEachStep() const
+    {
+        return mReformDofsAtEachStep;
+    }
+
+    /**
+     * @brief This method returns the flag mStiffnessMatrixIsBuilt
+     * @return The flag that indicates if the stiffness matrix is built
+     */
+    bool GetStiffnessMatrixIsBuilt() const
+    {
+        return mStiffnessMatrixIsBuilt;
+    }
+
+    /**
+     * @brief This method returns the residual norm
+     * @return The residual norm
+     */
+    double GetResidualNorm()
+    {
+        auto p_rhs = mLinearSystemContainer.pRhs;
+        KRATOS_ERROR_IF(p_rhs == nullptr) << "Right hand side vector is not set. Residual cannot be computed." << std::endl;
+        if (p_rhs->size() != 0) {
+            return p_rhs->Norm();
+        } else {
+            return 0.0;
+        }
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ImplicitStrategy";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    LinearSolverPointerType mpLinearSystemSolver = nullptr; /// Pointer to the linear solver
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief This method assigns settings to member variables
+     * @param ThisParameters Parameters that are assigned to the member variables
+     */
+    void AssignSettings(const Parameters ThisParameters)
+    {
+        // BaseType::AssignSettings(ThisParameters); // TODO: Once we have a base class
+        mEchoLevel = ThisParameters["echo_level"].GetInt();
+        mComputeReactions = ThisParameters["compute_reactions"].GetBool();
+        mReformDofsAtEachStep = ThisParameters["reform_dofs_at_each_step"].GetBool();
+
+        // Saving the scheme
+        if (ThisParameters["scheme_settings"].Has("name")) {
+            KRATOS_ERROR << "IMPLEMENTATION PENDING IN CONSTRUCTOR WITH PARAMETERS" << std::endl;
+        }
+    }
+
+    /**
+     * @brief This method returns the components of the system of equations depending of the echo level
+     */
+    virtual void EchoInfo()
+    {
+        const auto& r_A = mLinearSystemContainer.pLhs;
+        const auto& r_b = mLinearSystemContainer.pRhs;
+        const auto& r_dx = *mLinearSystemContainer.pDx;
+
+        if (GetEchoLevel() == 3) { //if it is needed to print the debug info
+            KRATOS_INFO("LHS") << "LHS = " << r_A << std::endl;
```

Comment:
A priori the label should be the class name.

Reply:
Yep. Corrected.